### DISCUSSION TITLE
Cannot convert json files to tcx

### DISCUSSION THREAD

**Discussion by polymath53** [CTHRU/Huawei-TCX-Converter#76] [Q&A]
Thank you! This is a useful tool for Huawei Health app users.

I requested my data twice in the Huawei Health app and successfully unzipped the files. However, I noticed that the 'motion path detail data.json' file size is only 1 kb. I tried using both Hitrava web and local editions, but neither could convert the JSON file to TCX. I am wondering if unzipping other files or finding alternative solutions would be helpful?

<img width="931" alt="hitrava web" src="https://github.com/CTHRU/Hitrava/assets/118784988/986db3fa-eb73-4bce-92d6-8505194c07c1">

<img width="733" alt="屏幕截图 2023-12-21 150513" src="https://github.com/CTHRU/Hitrava/assets/118784988/d139a200-59d4-4a0b-afd0-dab2a075af00">

<img width="816" alt="motion path" src="https://github.com/CTHRU/Hitrava/assets/118784988/8ac9e702-700a-43f0-a691-a5f8390b0eb2">

**Comment by CTHRU** [comment]
The motion path detail data.json file is empty. Probably the workout sessions didn't synchronize from your phone to the Huawei Health cloud servers. I suggest to check the privacy settings under 'Me' (bottom right icon on the main screen) > Privacy Management. You should have options there to synchronize your workout, personal information, and health data with the cloud (I don't have the English version, so I can't replicate the actual name of the options). Make sure the workout option is enabled. Then go to the main screen and drag from the top of the screen down. This should start the sync process. Good luck.

**Reply by polymath53** [reply]
Thanks! I think the data.json file has no motion path details because Huawei exports only sessions **_from the last year by default_**, and I have not used Huawei to record my runs during that time. Huawei is currently processing my request to export all of my records. [A post](https://www.fitconverter.com/convert/export/huawei5) suggests that Huawei can provide records from the past five years. The post also offers a tool to convert files, but [the website](https://www.fitconverter.com/) is in Chinese and requires payment.
